### ISSUE TITLE
Sort flatten search result alphabetically

### ISSUE THREAD

**Issue by bill-zhongjie-li** [dowjones/react-dropdown-tree-select#302]
In the case of searching the dropdown tree list, by default the search result is flattened. Can we sort the search result alphabetically, or provide an option to sort them alphabetically?

**Comment by mrchief** [comment]
It preserves the tree order. Sorting them alphabetically could be done but the UX could become confusing. Doing it via option adds some complexity especially when you consider `keepTreeOnSearch`. I don't see an easy way out but would be open to looking at a PR if it doesn't add much complexity. 

The next version should make it easier though as you'd be able to bring in your own renderer where you could effectively sort the tree before rendering.

**Comment by hamzhi** [comment]
Hi there, @bill-zhongjie-li you can sort data and children before put it into component, it's not dificult. But I have another problem. @mrchief  when I trying to search word 'test' and parent and child includes this word, then child move in the end.
https://codesandbox.io/s/react-dropdown-tree-select-ordi4

![image](https://user-images.githubusercontent.com/51838064/68209540-43fc3180-ffdc-11e9-984a-52b67e3b2aab.png)
![image](https://user-images.githubusercontent.com/51838064/68209561-51192080-ffdc-11e9-8d2b-c0f2e4880d14.png)

**Comment by mrchief** [comment]
That shouldn't happen. Do you have a codesandbox that reproduces this?

**Comment by hamzhi** [comment]
Of course, see my post above, or I'll just duplicate the link here https://codesandbox.io/s/react-dropdown-tree-select-ordi4

**Comment by mrchief** [comment]
Oh, sorry for missing that. Thanks! I'll take a look.

**Comment by ramijaschek** [comment]
I encounter a similar issue with search on the tree changing the order:

![image](https://user-images.githubusercontent.com/54894242/209664742-6f56bf1c-50ed-4a57-a278-8eae03f9e790.png)

vs.

![image](https://user-images.githubusercontent.com/54894242/209664819-5c23d819-6b9a-44a2-82c3-a47d3c643c78.png)
